### ISSUE TITLE
Invoke-RestMethod Error

### ISSUE THREAD

**Issue by joeyquack** [dfinke/PowerShellAI#169]
Hi Doug,
I am getting the following error message, I hope yo ucan help. I have been following the guide from this video:

https://www.youtube.com/watch?v=Ky9qcElTIRo

Attached is the error message

Thanks
<img width="462" alt="capture" src="https://github.com/dfinke/PowerShellAI/assets/138826277/3e09ff37-82b0-431c-b624-3f3f391ec9b7">

**Comment by dfinke** [comment]
You need to get on the waitlist for GPT-4.

Try two things:

See what models are availabee to your account.
```powershell
Get-OpenAIModel gpt*
```

You should have this:
```
Set-ChatSessionOption -model gpt-3.5-turbo
```

`New-Chat` sets the steerability of the model so.

```powershell
new-chat 'you are a expert PowerShell bot'
```

Finally:

```powershell
chat 'list the even numbers between 1 and 10'
```

**Comment by joeyquack** [comment]
Hi Doug,
Thanks for sending this info over. I already have chatgpt 4, I pay the monthly subscription. Can you advise any further before I try what you suggested?

Thanks

Sent from my iPhone

On 6 Jul 2023, at 11:06 pm, Doug Finke ***@***.***> wrote:

﻿

You need to get on the waitlist for GPT-4.

Try two things:

See what models are availabee to your account.

Get-OpenAIModel gpt*

You should have this:

Set-ChatSessionOption -model gpt-3.5-turbo


New-Chat sets the steerability of the model so.

new-chat 'you are a expert PowerShell bot'

Finally:

chat 'list the even numbers between 1 and 10'

**Comment by dfinke** [comment]
That is very weird. The error indicates you don't have it. I don't know.

Can you post the outputof this `Get-OpenAIModel gpt*`

**Comment by whytheq** [comment]
Is paying the GPT4 general subscription the same as access to the GPT4 API ? I think they are separate.

**Comment by joeyquack** [comment]
![Screenshot 2023-07-07 at 10 17 40](https://github.com/dfinke/PowerShellAI/assets/138826277/d69dbca2-e4a3-43c5-b607-05e6153990e5)

Hi Doug,
This is the output I received.
Thanks,
Joey

**Comment by dfinke** [comment]
I think they just made GPT-4 generally available. 

Before you had to pay and get on the waitlist.

See if you have it now, if not, it may still be rolling out.

**Comment by joeyquack** [comment]
I have resolved this. Feel free to close the ticket, thank you for your help all!

**Comment by dfinke** [comment]
How was it resolved?

**Comment by joeyquack** [comment]
I have purchased the ChatGPT 4 upgrade, but I have not yet configured the billing for the ChatGPT API.
________________________________

**Comment by RStadlmair** [comment]
Hi ! Same issue here. 
Yesterday (23/08/23) i purchased a ChatGPT Subscription for USD 20/month.
Unfortunately on "platform.openai.com", this doesnt show up.

Maybe my fault i probably should have RTFM´ed, i will now change to OpenAI platform mayment.

**Comment by dfinke** [comment]
@RStadlmair thanks for the update. 

The $20 ChatGPT Plus is worth keeping for a couple months (easily cancelled) and kicking the tires on. You get access to new features. The newest being the Code Interpreter.

Here are some livestreams I've done on the Code Interpreter:

https://www.youtube.com/playlist?list=PL5uoqS92stXibiBR1J05x4jPzz5Jv4QQ2

**Comment by mynamejeff22** [comment]
I am also having this issue with free version, using the instructions above:

> You need to get on the waitlist for GPT-4.
> 
> Try two things:
> 
> See what models are availabee to your account.
> 
> ```powershell
> Get-OpenAIModel gpt*
> ```
> 
> You should have this:
> 
> ```
> Set-ChatSessionOption -model gpt-3.5-turbo
> ```
> 
> `New-Chat` sets the steerability of the model so.
> 
> ```powershell
> new-chat 'you are a expert PowerShell bot'
> ```
> 
> Finally:
> 
> ```powershell
> chat 'list the even numbers between 1 and 10'
> ```

![image](https://github.com/dfinke/PowerShellAI/assets/139219325/ebb60c1a-17b4-43f8-828c-b34e297b2f36)

**Comment by joeyquack** [comment]
> @joeyquack could you say a bit more?
> 
> > The issue stems from not having an API set up

Yes, your API needs to be configured via this link https://platform.openai.com/account/api-keys

Also billing needs to be configured and set up https://platform.openai.com/account/billing/overview
